instrumentation/opentelemetry-annotations-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/otelannotations/async/Jdk8MethodStrategy.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.otelannotations.async;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.tracer.BaseTracer;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+enum Jdk8MethodStrategy implements MethodSpanStrategy {
+  INSTANCE;
+
+  @Override
+  public boolean supports(Class<?> returnType) {
+    return returnType == CompletionStage.class || returnType == CompletableFuture.class;
+  }
+
+  @Override
+  public Object end(BaseTracer tracer, Context context, Class<?> returnType, Object result) {
+    if (result instanceof CompletableFuture) {
```

Comment:
to avoid problem if method is declared to return a subclass of `CompletableFuture`... although that's not possible due to `supports` implementation above... still, feels a bit safer in case this pattern is extended/copied in the future
```suggestion
    if (returnType == CompletableFuture.class) {
```

Reply:
I've added a double check by having `end` call `supports` on the `returnType`.  I left the use of `instanceof` as once the method signature is checked it is safe to deal with subclasses, e.g. if the method is declared to return `CompletionStage<T>` but actually returns `CompletableFuture<T>`.